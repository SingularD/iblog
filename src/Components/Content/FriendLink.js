import React from 'react'

import NavBar from './NavBar'


class FriendLink extends React.Component{
  render() {
    return (
      <React.Fragment>
        <NavBar />
        <div className="col-12 col-lg-9 row friend-link">
          <h1>尚未开放，请期待...</h1>
        </div>
      </React.Fragment>
    )
  }
}

export default FriendLink
