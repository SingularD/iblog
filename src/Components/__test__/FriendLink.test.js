import React from 'react'
import { shallow } from "enzyme";
import FriendLink from '../Content/FriendLink'

describe("测试友链是否渲染", () => {
  let wrapper = shallow(<FriendLink />)
  it("匹配快照", () => {
    expect(wrapper).toMatchSnapshot()
  })
  it("测试渲染", () => {
    expect(wrapper.find('.friend-link h1').text()).toEqual('尚未开放，请期待...')
  })
})
