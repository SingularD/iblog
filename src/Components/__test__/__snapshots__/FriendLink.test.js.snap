// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试友链是否渲染 匹配快照 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FriendLink />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <NavBar />,
        <div
          className="col-12 col-lg-9 row friend-link"
        >
          <h1>
            尚未开放，请期待...
          </h1>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            尚未开放，请期待...
          </h1>,
          "className": "col-12 col-lg-9 row friend-link",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "尚未开放，请期待...",
          },
          "ref": null,
          "rendered": "尚未开放，请期待...",
          "type": "h1",
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavBar />,
          <div
            className="col-12 col-lg-9 row friend-link"
          >
            <h1>
              尚未开放，请期待...
            </h1>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              尚未开放，请期待...
            </h1>,
            "className": "col-12 col-lg-9 row friend-link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "尚未开放，请期待...",
            },
            "ref": null,
            "rendered": "尚未开放，请期待...",
            "type": "h1",
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
